Sources/CvPlayer.cpp

```diff
@@ -5982,39 +5953,32 @@ void CvPlayer::updatePlunder(int iChange, bool bUpdatePlotGroups)
 
 void CvPlayer::updateTimers()
 {
-	CvSelectionGroup* pLoopSelectionGroup;
-	int iLoop;
-
-	for(pLoopSelectionGroup = firstSelectionGroup(&iLoop); pLoopSelectionGroup; pLoopSelectionGroup = nextSelectionGroup(&iLoop))
+	for (CvPlayer::group_iterator groupItr = beginGroups(); groupItr != endGroups(); ++groupItr)
 	{
-		pLoopSelectionGroup->updateTimers(); // could destroy the selection group...
+		(*groupItr)->updateTimers(); // could destroy the selection group...
 	}
 
 	// if a unit was busy, perhaps it was not quite deleted yet, give it one more try
 	if (getNumSelectionGroups() > getNumUnits())
 	{
-		for(pLoopSelectionGroup = firstSelectionGroup(&iLoop); pLoopSelectionGroup; pLoopSelectionGroup = nextSelectionGroup(&iLoop))
+		for (CvPlayer::plot_group_iterator plotGroupItr = beginPlotGroups(); plotGroupItr != endPlotGroups(); ++plotGroupItr)
 		{
-			pLoopSelectionGroup->doDelayedDeath(); // could destroy the selection group...
+			(*groupItr)->doDelayedDeath(); // could destroy the selection group...
 		}
 	}
```

Comment:
Do I need to use the other iterator here?

Reply:
`CvPlayer::group_iterator`